openedx_learning/apps/authoring/publishing/models/entity_list.py

```diff
@@ -0,0 +1,60 @@
+"""
+Entity List models
+"""
+from django.db import models
+
+from .publishable_entity import PublishableEntity, PublishableEntityVersion
+
+
+class EntityList(models.Model):
+    """
+    EntityLists are a common structure to hold parent-child relations.
+
+    EntityLists are not PublishableEntities in and of themselves. That's because
+    sometimes we'll want the same kind of data structure for things that we
+    dynamically generate for individual students (e.g. Variants). EntityLists are
+    anonymous in a sense–they're pointed to by ContainerVersions and
+    other models, rather than being looked up by their own identifiers.
+    """
+
+
+class EntityListRow(models.Model):
+    """
+    Each EntityListRow points to a PublishableEntity, optionally at a specific
+    version.
+
+    There is a row in this table for each member of an EntityList. The order_num
+    field is used to determine the order of the members in the list.
+    """
+
+    entity_list = models.ForeignKey(EntityList, on_delete=models.CASCADE)
+
+    # This ordering should be treated as immutable–if the ordering needs to
+    # change, we create a new EntityList and copy things over.
+    order_num = models.PositiveIntegerField()
```

Comment:
I think this is missing a unique constraint on (entity_list, order_num), so we don't get duplicates in race conditions.

Reply:
I was thinking it doesn't really matter if `order_num` is not unique, but you're right - if it's not unique, it likely indicates a race condition and we should raise an exception rather than ignoring.

Added in e6334fb0a79598887e9945d85cd1d6a17c954da5